src/UniversalRewardsDistributor.sol

```diff
@@ -17,238 +18,163 @@ import {MerkleProof} from "@openzeppelin/utils/cryptography/MerkleProof.sol";
-        if (newTimelock < timelockOf[distributionId]) {
-            PendingRoot memory pendingRoot = pendingRootOf[distributionId];
+    function updateTimelock(uint256 newTimelock) external onlyOwner {
+        if (newTimelock < timelock) {
+            PendingRoot memory pendingRootMem = pendingRoot;
             require(
-                pendingRoot.submittedAt == 0 || pendingRoot.submittedAt + timelockOf[distributionId] <= block.timestamp,
+                pendingRootMem.submittedAt == 0 || pendingRootMem.submittedAt + timelock <= block.timestamp,
                 ErrorsLib.TIMELOCK_NOT_EXPIRED
             );
         }
 
-        timelockOf[distributionId] = newTimelock;
-        emit TimelockUpdated(distributionId, newTimelock);
+        timelock = newTimelock;
+        emit EventsLib.TimelockUpdated(newTimelock);
     }
 
     /// @notice Updates the root updater of a given distribution.
-    /// @param distributionId The distributionId of the merkle tree distribution.
     /// @param updater The new root updater.
     /// @param active Whether the root updater should be active or not.
-    function updateRootUpdater(uint256 distributionId, address updater, bool active)
-        external
-        onlyOwner(distributionId)
-    {
-        isUpdaterOf[distributionId][updater] = active;
-        emit RootUpdaterUpdated(distributionId, updater, active);
+    function updateRootUpdater(address updater, bool active) external onlyOwner {
+        isUpdater[updater] = active;
+        emit EventsLib.RootUpdaterUpdated(updater, active);
     }
 
     /// @notice Revokes the pending root of a given distribution.
-    /// @param distributionId The distributionId of the merkle tree distribution.
     /// @dev This function can only be called by the owner of the distribution at any time.
-    function revokePendingRoot(uint256 distributionId) external onlyOwner(distributionId) {
-        require(pendingRootOf[distributionId].submittedAt != 0, ErrorsLib.NO_PENDING_ROOT);
-
-        delete pendingRootOf[distributionId];
-        emit PendingRootRevoked(distributionId);
-    }
+    function revokePendingRoot() external onlyOwner {
+        require(pendingRoot.submittedAt != 0, ErrorsLib.NO_PENDING_ROOT);
 
-    function setDistributionOwner(uint256 distributionId, address newOwner) external onlyOwner(distributionId) {
-        ownerOf[distributionId] = newOwner;
-        emit DistributionOwnerSet(distributionId, msg.sender, newOwner);
+        delete pendingRoot;
+        emit EventsLib.PendingRootRevoked();
     }
 
-    function _forceUpdateRoot(uint256 distributionId, bytes32 newRoot, bytes32 newIpfsHash) internal {
-        rootOf[distributionId] = newRoot;
-        ipfsHashOf[distributionId] = newIpfsHash;
-        delete pendingRootOf[distributionId];
-        emit RootUpdated(distributionId, newRoot, newIpfsHash);
+    function setDistributionOwner(address newOwner) external onlyOwner {
```

Comment:
Beyond the scope of this PR but we should either have it changed here or have a dedicated issue: we should be consistent in the usage of `set` or `update`. I'm fine with both, have a preference for `set` but most of all I require to be consistent!

Reply:
I'm still trying to figure that out. IMO `setRoot` is not correct. You are not setting the root (the root is set at the moment the root updater submits a pending root), but you are updating the root with a proposed one (potentially by someone else, some time ago) 

set = you are defining a new onchain value
update = you are just triggering an on-chain change, without necessarily providing any data.